packages/tools/src/utilities/segmentation/getSegmentAtWorldPoint.ts

```diff
@@ -0,0 +1,146 @@
+import { cache, utilities } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { SegmentationRepresentations } from '../../enums';
+import {
+  findSegmentationRepresentationByUID,
+  getSegmentation,
+} from '../../stateManagement/segmentation/segmentationState';
+import {
+  LabelmapSegmentationData,
+  LabelmapSegmentationDataStack,
+  LabelmapSegmentationDataVolume,
+} from '../../types/LabelmapTypes';
+import { isVolumeSegmentation } from '../../tools/segmentation/strategies/utils/stackVolumeCheck';
+import {
+  ContourSegmentationAnnotation,
+  ContourSegmentationData,
+} from '../../types';
+import { getAnnotation } from '../../stateManagement';
+import { isPointInsidePolyline3D } from '../math/polyline';
+
```

Comment:
Could we get some unit tests on this?  I'd particularly like to see tests that check that if the slice is different/same as the expected data, we get the right results.

Reply:
Oh, and data inside/outside the range of available indices should also be tested.

Reply:
I believe it would be great to include that, but unfortunately, the budget for this ticket has been used up. I have added it to the to-do list.